modules/protocol/inspircd.c

```diff
@@ -134,18 +183,21 @@ inspircd_next_matching_ban(struct channel *c, struct user *u, int type, mowgli_n
 		if (cb->type != type)
 			continue;
 
-		if ((!match(cb->mask, hostbuf) || !match(cb->mask, realbuf) || !match(cb->mask, ipbuf)) || !match_cidr(cb->mask, ipbuf))
-			return n;
+		if ((!match(mask, hostbuf) || !match(mask, realbuf) || !match(mask, ipbuf)) || !match_cidr(mask, ipbuf))
+			return invert ? NULL : n;
```

Comment:
hrm, correct me if I'm misunderstanding things – it looks like you're checking for a normal match, exiting early if bannegate was used on the banmask. It seems like we'd need to invert the matching logic instead (though note that if a ban doesn't match against a user's `nick!user@host`, that'll only be an inverted match if it's not an extban instead)